### ISSUE TITLE
Definition Tooltip: AVT 1 violation

### ISSUE THREAD

**Issue by elizabethsjudd** [carbon-design-system/carbon#3099]
## Environment

> Operating system

Mac OS X

> Browser

Safari

> Automated testing tool and ruleset

AAT 2.4.1: https://aat.w3ibm.mybluemix.net/token/c1e59337-69c6-45fb-9ff0-6379259a1405/67022795-93d7-4142-b7e3-712ea042711b/archives/2019FebDeploy//doc/w3/help/en-US/idhi_accessibility_check_g1202.html

> Assistive technology used to verify

VoiceOver

## Detailed description

Button's can't use `aria-label` as it overwrites the contents of the button so screen reader user is not getting the text that displays in the trigger

> What version of the Carbon Design System are you using?

10.3.1

> What did you expect to happen?

Screen reader users would be able to access all content.

> What happened instead?

Screen reader users only got the tooltip content

> What WCAG 2.1 checkpoint does the issue violate?

## Steps to reproduce the issue

1. Use a screen reader on Carbon's demo of the definition tooltip

## Additional information
You need to use the `aria-describedby` attribute and point to a different element that contains the content of the tooltip separately from the trigger
<img width="1566" alt="Screen Shot 2019-06-18 at 12 45 25 PM" src="https://user-images.githubusercontent.com/1275779/59703378-be601c80-91c7-11e9-9702-367ff4fbe16b.png">

**Comment by elizabethsjudd** [comment]
@carbon-design-system/developers

RFC: There are a couple different options we have for implementing this and would like to get Carbon's input before proceeding

1. Implement similar to #3115 
Pros: uses less Sass as it can share with icon tooltip
Cons: The contents of the button can get fairly lengthy (readability of code)
```
<div class="{{@root.prefix}}--tooltip--definition">
  <button
    class="{{@root.prefix}}--tooltip__trigger {{@root.prefix}}--tooltip--a11y {{@root.prefix}}--tooltip__trigger--definition {{@root.prefix}}--tooltip--bottom {{@root.prefix}}--tooltip--align-start">
    Definition Tooltip (start aligned)
    <span class="{{@root.prefix}}--tooltip__content">Brief description of the dotted, underlined word above.</span>
  </button>
</div>
```
2. Use tooltip role
Pros: The contents of the tooltip are separate from the trigger making the HTML more readable
Cons: Additional sass is required to support the different HTML structure
```
<div class="{{@root.prefix}}--tooltip--definition">
  <button
    class="{{@root.prefix}}--tooltip__trigger {{@root.prefix}}--tooltip--a11y {{@root.prefix}}--tooltip__trigger--definition {{@root.prefix}}--tooltip--bottom {{@root.prefix}}--tooltip--align-start"
    aria-describedby="start-example">
    Definition Tooltip (start aligned)
  </button>
  <div id="start-example" role="tooltip" class="{{@root.prefix}}--tooltip__content">Brief description of the dotted, underlined word above.</div>
</div>
```

**Comment by elizabethsjudd** [comment]
Also, all of the demos of these tooltip use a button as the trigger but according to the usage docs (https://www.carbondesignsystem.com/components/tooltip/usage#definition-tooltip) it's often not a button.  The use of a button actually changes the behavior of Tooltip (it can get in a persistent state and remain open if the button is clicked but if it was applied to static text it wouldn't do that).  Can we get some clarification from @carbon-design-system/design around what's expected for this component?

**Comment by tw15egan** [comment]
cc @carbon-design-system/design

**Comment by snidersd** [comment]
This issue is related to issue #2736.
